File: average-word-length/AverageWordLength.c
#include <stdio.h>
#include <math.h>
#define MAX 5000

int main() {
  char essay[MAX];
  
  int words = 0;
  int characters = 0;
  char previous = 0;
  
  for (char *word = fgets(essay, MAX, stdin); *word; previous = *(word++))
    if (isalpha(*word)) ++characters;
    else if (isalpha(previous)) ++words;
    
  if (isalpha(previous)) ++words;
  printf("%d", (int)ceil(characters / (double)words));
  
  return 0;
}

File: average-word-length/AverageWordLength.java
import java.util.Scanner;

public class AverageWordLength
{
    public static void main(String[] args) {
        final var input = new Scanner(System.in);
        final var essay = input.nextLine();
        final int words = essay.split(" ").length;
        final int chars = essay.replaceAll("[^a-zA-Z]", "").length();
        
        final int average = (int) java.lang.Math.ceil((double) chars / words);
        
        System.out.print(average);
    }
}
